src/main/java/randoop/resource/generator/MethodListGen.java

```diff
@@ -0,0 +1,173 @@
+          }
+          sideEffectMethodWriter.flush();
+        }
+      } else {
+        System.out.println("No nonDet methods found. Not writing to file.");
+      }
+    } catch (IOException e) {
+      e.printStackTrace();
+      System.exit(1);
+    }
+  }
+
+  private static List<String> walkFilesForAnnotation(
+      Path root, String fileType, List<String> qualifyingAnnotations) throws IOException {
+    List<String> annotatedMethods = new ArrayList<>();
+    // Recursively walk the file directory
+    try (Stream<Path> paths = Files.walk(root).filter(Files::isRegularFile)) {
+      paths.forEach(
+          filePath -> {
+            if (filePath.toString().endsWith(fileType)) {
+              try {
+                annotatedMethods.addAll(readFile(filePath, fileType, qualifyingAnnotations));
+              } catch (IOException ex) {
+                throw (new RuntimeException(ex));
+              }
+            }
+          });
+    }
+    return annotatedMethods;
+  }
+
+  private static List<String> readFile(
+      Path filePath, String fileType, List<String> qualifyingAnnotations) throws IOException {
+    List<String> annotatedMethods = new ArrayList<>();
+
+    // Invoke the parser for the specified class file
+    AScene scene = null;
+    if (fileType.equals(CLASS_EXT)) {
+      scene = new AScene();
+      ClassFileReader.read(scene, filePath.toString());
+    } else { // JAIF_EXT
+      IndexFileSpecification ifs = new IndexFileSpecification(filePath.toString());
+      scene = ifs.getScene();
+      ifs.parse();
+    }
+
+    assert (scene != null);
+
+    // Look through each class and its corresponding methods.
+    for (Map.Entry<String, AClass> entry : scene.classes.entrySet()) {
+      AClass aclass = entry.getValue();
+      for (Map.Entry<String, AMethod> m : aclass.methods.entrySet()) {
+        // Check annotations for the method
+        Collection<Annotation> annotationLocation =
+            fileType.equals(CLASS_EXT)
+                ? m.getValue().returnType.tlAnnotationsHere
+                : m.getValue().tlAnnotationsHere;
+        for (Annotation a : annotationLocation) {
+          if (qualifyingAnnotations.contains(a.def.name)) {
+            String fullyQualifiedName = getFullyQualifiedName(aclass, m.getValue().methodName);
+            if (!fullyQualifiedName.contains("<init>") && !fullyQualifiedName.contains("$")) {
+              annotatedMethods.add(fullyQualifiedName);
+            }
+            break;
+          }
+        }
+      }
+    }
+    return annotatedMethods;
+  }
+
+  private static String getFullyQualifiedName(AClass aclass, String JVMLmethodSignature) {
+    // Parse out the arguments to feed into plume.
+    int arglistStartIndex = JVMLmethodSignature.indexOf("(");
+    int arglistEndIndex = JVMLmethodSignature.indexOf(")") + 1;
+    String methodName = JVMLmethodSignature.substring(0, arglistStartIndex);
+
+    String fullyQualifiedMethodName = aclass.className;
```

Comment:
Why is this called `fullyQualified**Method**Name`?  It seems to be a class name.